candi/bashible/bundles/centos/bootstrap.sh.tpl

```diff
@@ -78,8 +78,12 @@ bb-rp-get-token() {
   fi
 
   AUTH_HEADER="$(curl --retry 3 -k -sSLi "${SCHEME}://${REGISTRY_ADDRESS}/v2/" | grep -i "www-authenticate")"
-  AUTH_REALM="$(awk -F "," '{split($1,s,"\""); print s[2]}' <<< "${AUTH_HEADER}")"
-  AUTH_SERVICE="$(awk -F "," '{split($2,s,"\""); print s[2]}' <<< "${AUTH_HEADER}" | sed "s/ /+/g")"
+  AUTH_REALM="$(grep -oE 'Bearer realm="http[s]{0,1}://[a-z0-9\.\:\/\-]+"' <<< ${AUTH_HEADER} | cut -d '"' -f2)"
+  AUTH_SERVICE="$(grep -oE 'service="[[:print:]]+"' <<< "${AUTH_HEADER}" | cut -d '"' -f2 | sed 's/ /+/g')"
+  if [ -z ${AUTH_REALM} ]; then
+    >&2 echo "cannot parse registry auth header ${AUTH_HEADER}"
```

Comment:
(duplicate) There should be description, what's wrong with the header. Like 'couldn't find Bearer realm parameter, consider enabling docker realm in your registry'.

Reply:
done